modules/registration/registration_api.rb

```diff
@@ -0,0 +1,34 @@
+require 'registration/proxy_request'
+
+class Proxy::Registration::Api < ::Sinatra::Base
+  get '/' do
+    response = Proxy::Registration::ProxyRequest.new.global_register(request)
+    handle_response(response)
+  rescue StandardError => e
+    logger.error "Error when rendering Global Registration Template: #{e}"
+    render_error
+  end
+
+  post '/' do
+    response = Proxy::Registration::ProxyRequest.new.host_register(request)
+    handle_response(response)
+  rescue StandardError => e
+    logger.error "Error when rendering Host Registration Template: #{e}"
+    render_error
+  end
+
+  private
+
+  def handle_response(response)
+    if response.code == '200'
+      response.body
```

Comment:
All 2xx codes are considered valid. On a HTTP POST you would expect a HTTP 201 if a resource was created.